.github/workflows/main.yml

```diff
@@ -1,44 +1,63 @@
-    - name: Build the testing docker image
-      run: docker build -t pltest .
-    - name: Start the container
-      run: docker run -td --name=test_container pltest /bin/bash
-    - name: Run the JavaScript linter
-      run: docker exec test_container /PrairieLearn/docker/lint_js.sh
-    - name: Run the Python linter
-      run: docker exec test_container /PrairieLearn/docker/lint_python.sh
-    - name: Run typechecker
-      run: docker exec test_container /PrairieLearn/docker/typecheck.sh
-    - name: Run the PrairieLib tests
-      run: docker exec test_container /PrairieLearn/docker/test_prairielib.sh
-    - name: Run the grader_host tests
-      run: docker exec test_container /PrairieLearn/docker/test_grader_host.sh
-    - name: Run the JavaScript tests
-      run: docker exec test_container /PrairieLearn/docker/test_js.sh
-    - name: Run the Python tests
-      run: docker exec test_container /PrairieLearn/docker/test_python.sh
-    - name: Copy code-coverage data out of container
-      run: docker cp test_container:/PrairieLearn/coverage coverage
-    - name: Directory listing for debug
-      run: ls -lR
-    - name: Head of lcov.info for debug
-      run: head -1000 coverage/lcov.info
-    - name: Send code-coverage data to Coveralls
-      uses: coverallsapp/github-action@master
-      with:
-        github-token: ${{ secrets.github_token }}
-      continue-on-error: true
+      - uses: actions/checkout@v2
+        with:
+          # Checkout the PR HEAD commit instead of merge commit
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Build the plbase docker image if needed
+        run: sh ./tools/build-plbase-if-needed.sh
+      - name: Build the testing docker image
+        run: docker build -t pltest .
+      - name: Start the container
+        run: docker run -td --name=test_container pltest /bin/bash
+      - name: Run the JavaScript linter
+        run: docker exec test_container /PrairieLearn/docker/lint_js.sh
+      - name: Run the Python linter
+        run: docker exec test_container /PrairieLearn/docker/lint_python.sh
+      - name: Run typechecker
+        run: docker exec test_container /PrairieLearn/docker/typecheck.sh
+      - name: Run the PrairieLib tests
+        run: docker exec test_container /PrairieLearn/docker/test_prairielib.sh
+      - name: Run the grader_host tests
+        run: docker exec test_container /PrairieLearn/docker/test_grader_host.sh
+      - name: Run the JavaScript tests
+        run: docker exec test_container /PrairieLearn/docker/test_js.sh
+      - name: Run the Python tests
+        run: docker exec test_container /PrairieLearn/docker/test_python.sh
+      - name: Copy code-coverage data out of container
+        run: docker cp test_container:/PrairieLearn/coverage coverage
+      - name: Directory listing for debug
+        run: ls -lR
+      - name: Head of lcov.info for debug
+        run: head -1000 coverage/lcov.info
```

Comment:
I think we don't need this anymore. Some time ago we were getting action failures from coveralls (and we added this code to debug), but they seem to have been fixed by upstream.

Reply:
Gotcha! I can drop it here.